packages/starlight/user-components/Tabs.astro

```diff
@@ -159,26 +220,42 @@ const { html, panels } = processPanels(panelHtml);
 			newTab.setAttribute('aria-selected', 'true');
 			if (shouldSync) {
 				newTab.focus();
-				StarlightTabs.#syncTabs(this, newTab.innerText);
+				const label = StarlightTabs.getTabLabel(newTab);
+				if (!label) return;
+				StarlightTabs.#syncTabs(this, label);
 				window.scrollTo({
 					top: window.scrollY + (this.getBoundingClientRect().top - previousTabsOffset),
 				});
+				this.#persistSyncedTabs(label);
```

Comment:
Small nit that maybe it makes more sense to move this (and probably the `getTabLabel()`) inside `#syncTabs()` directly so that the logic is always `#syncTabs() => #persistSyncedTabs()` if we ever wanted to sync tabs from somewhere else for some reason, but it’s not crucial.

Reply:
Makes total sense now with a more advanced prototype. Thanks again for the early review :raised_hands:

Reply:
Refactor looks much more readable to me now too — nice work!